File: apps/mobile/ios/Runner/video_picker/VideoPicker.swift
import Foundation
import UniformTypeIdentifiers

/// Protocol for video picker operations.
/// Equivalent to Android's VideoPicker interface.
protocol VideoPicker {
    
    /// The delegate to receive picker results
    var delegate: VideoPickerDelegate? { get set }
    
    /// Opens the system document picker for multiple video selection.
    /// Results are returned via the delegate.
    func pickVideos()
    
    /// Opens the system document picker for single video selection.
    /// Results are returned via the delegate.
    func pickSingleVideo()
}

/// Delegate protocol for receiving video picker results.
protocol VideoPickerDelegate: AnyObject {
    
    /// Called when multiple videos are picked.
    /// - Parameter urls: Array of file URL strings, or nil if cancelled
    func didPickVideos(_ urls: [String]?)
    
    /// Called when a single video is picked.
    /// - Parameter url: File URL string, or nil if cancelled
    func didPickSingleVideo(_ url: String?)
}

File: apps/mobile/android/app/src/main/kotlin/open/fpvlabs/stera/video_picker/VideoPicker.kt
package open.fpvlabs.stera.video_picker

import android.content.Intent

/**
 * Interface for video picker operations.
 */
interface VideoPicker {

    /**
     * Opens the system file picker for multiple video selection.
     * Results are returned via the callback set in the handler.
     */
    fun pickVideos()

    /**
     * Opens the system file picker for single video selection.
     * Results are returned via the callback set in the handler.
     */
    fun pickSingleVideo()

    /**
     * Handles the activity result from the picker.
     * @param requestCode The request code from onActivityResult
     * @param resultCode The result code from onActivityResult
     * @param data The intent data from onActivityResult
     * @return Pair of (isHandled, result) - result is List<String>? for multiple, String? for single
     */
    fun handleActivityResult(requestCode: Int, resultCode: Int, data: Intent?): Pair<Boolean, Any?>
}